src/poetry/console/commands/shell.py

```diff
@@ -21,10 +20,7 @@ def handle(self) -> int:
         from poetry.utils.shell import Shell
 
         # Check if it's already activated or doesn't exist and won't be created
-        venv_activated = strtobool(environ.get("POETRY_ACTIVE", "0")) or getattr(
-            sys, "real_prefix", sys.prefix
-        ) == str(self.env.path)
-        if venv_activated:
+        if self._is_venv_activated():
             self.line(
                 f"Virtual environment already activated: <info>{self.env.path}</>"
             )
```

Comment:
Not related to those changes, but the line after this one makes the method return `0`, so the command will have an exit code `0` in the shell.
Is this the correct thing to do? I would have expected the command to end up with an exit code `1` if the shell is already active.

Reply:
I think the intention is to be a soft no-op as opposed to a failure -- we could always revisit this later, but I don't think it's a major issue.